lib/service/Controller/Swap/SwapController.dart

```diff
@@ -85,12 +86,16 @@ class SwapController extends GetxController {
     List<EthereumAddress> path = [tokenAAddress, tokenBAddress];
     EthereumAddress to = await controller.credentials.extractAddress();
     String txString = "";
+    var bloc;
+    bloc = BuyDialogBloc;
 
     try {
       txString = await _aptRouter.swapExactTokensForTokens(
           tokenAAmount, amountOutMin, path, to, deadline.value,
           credentials: controller.credentials);
     } catch (e) {
+      bloc.add(OnErrorEvent());
+
```

Comment:
I would like to know why are we adding the bloc events in the swap controller?

Reply:
Indeed this is not something that should have reference to the `BuyDialogBloc`...